src/lib/formatters.js

```diff
@@ -9,3 +9,67 @@ export function dateTimeFormat(timestamp, options) {
 
   return new Intl.DateTimeFormat(undefined, chosenOptions).format(dateTime);
 }
+
+export function serializeFilters(searchParams, updateFilter) {
+  Object.keys(updateFilter).forEach((key) => {
+    let searchParamVal = null;
+
+    if (
+      key === 'id' ||
+      key === 'viewType' ||
+      key === 'type' ||
+      key === 'name'
+    ) {
+      searchParamVal = JSON.stringify([...updateFilter[key].exclude]);
+    } else if (key === 'timeStamp') {
+      searchParamVal = JSON.stringify(updateFilter[key]);
+    } else {
+      searchParamVal = updateFilter[key];
+    }
+
+    searchParams.set(key, searchParamVal);
+  });
+
+  return searchParams;
+}
+
+function getJSONParseVal(str) {
+  try {
+    return JSON.parse(str);
+  } catch (error) {
```

Comment:
I suppose that you've created this to work around errors. Did you mean to include (new?) unit tests that trigger the error, or is that part of follow-up work?

Reply:
Since you said to think about the edge cases (e.g. incorrect/missing input), I just put random search params in url and got an error from `JSON.parse`. This is the work arround for that. 
Should I add tests for this?

Reply:
Yes. There are more ways to break the functionality (e.g. something that is valid JSON but not having the expected type), but you'll discover that in tests.